main.tf

```diff
@@ -1 +1,100 @@
-#(empty)
+module "sessionkms" {
+  count            = var.kms_key_id == "" && (var.encrypt_session || var.cloudwatch_encryption_enabled || var.s3_encryption_enabled) ? 1 : 0
+  source           = "boldlink/kms/aws"
+  version          = "1.1.0"
+  description      = "AWS CMK for encrypting ssm session"
+  create_kms_alias = true
+  kms_policy       = local.kms_policy
```

Comment:
We can have the default kms key but we should allow for additional statements on the policy for customization.